proctor-common/src/main/java/com/indeed/proctor/common/RuleVerifyUtils.java

```diff
@@ -76,7 +77,8 @@ public static void verifyRule(
                 // Check identifiers in the AST and verify variable names
                 final Node undefinedIdentifier =
                         checkUndefinedIdentifier(root, elContext, absentIdentifiers);
-                if (undefinedIdentifier != null) {
+                if (undefinedIdentifier != null
```

Comment:
undefinedIdentifier is the first identified missing variable.
if UNITLESS_ALLOCATION_IDENTIFIER is missing, others wont be reported.

Reply:
Good catch didn't check that function.

Reply:
Moved logic to checkUndefinedIdentifier.